### PR TITLE
Write e2e test for PersistentVolumeStatus & PersistentVolumeClaimStatus Endpoints +6 Endpoints

### LINKED ISSUE #120891: Write e2e test for PersistentVolumeStatus & PersistentVolumeClaimStatus Endpoints +6 Endpoints
# Progress <code>[5/8]</code>

-   [x] APISnoop org-flow : [CoreV1PV-PVC-StatusTest.org](https://github.com/apisnoop/ticket-writing/blob/master/CoreV1PV-PVC-StatusTest.org)
-   [x] test approval issue : #120891
-   [x] test pr : #120892 
-   [x] two weeks soak start date : ~~12 Oct 2023~~ 17 Oct 2023 [testgrid-link](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&width=5&graph-metrics=test-duration-minutes&include-filter-by-regex=should%20apply%20changes%20to%20a%20pv/pvc%20status)
      -  [x] Flake fix #121243 
-   [ ] two weeks soak end date : ~~26 Oct 2023~~ 31 Oct 2023
-   [X] test promotion pr : #121331 
-  [X] remove endpoint from pending_eligible_endpoints.yaml #121332 

# Identifying an untested feature Using APISnoop

## Untested Endpoints

According to following APIsnoop query, there are three PersistentVolume and three PersistentVolumeClaim endpoints that are untested.

```sql-mode
    SELECT
      endpoint,
      path,
      kind
      FROM testing.untested_stable_endpoint
      where eligible is true
      and endpoint ilike '%PersistentVolume%'
      order by kind, endpoint
      limit 10;
```

```example
                        endpoint                      |                                path                                 |         kind
  ----------------------------------------------------+---------------------------------------------------------------------+-----------------------
   patchCoreV1PersistentVolumeStatus                  | /api/v1/persistentvolumes/{name}/status                             | PersistentVolume
   readCoreV1PersistentVolumeStatus                   | /api/v1/persistentvolumes/{name}/status                             | PersistentVolume
   replaceCoreV1PersistentVolumeStatus                | /api/v1/persistentvolumes/{name}/status                             | PersistentVolume
   patchCoreV1NamespacedPersistentVolumeClaimStatus   | /api/v1/namespaces/{namespace}/persistentvolumeclaims/{name}/status | PersistentVolumeClaim
   readCoreV1NamespacedPersistentVolumeClaimStatus    | /api/v1/namespaces/{namespace}/persistentvolumeclaims/{name}/status | PersistentVolumeClaim
   replaceCoreV1NamespacedPersistentVolumeClaimStatus | /api/v1/namespaces/{namespace}/persistentvolumeclaims/{name}/status | PersistentVolumeClaim
  (6 rows)

```

-   <https://apisnoop.cncf.io/1.28.0/stable/core/patchCoreV1NamespacedPersistentVolumeClaimStatus>
-   <https://apisnoop.cncf.io/1.28.0/stable/core/readCoreV1NamespacedPersistentVolumeClaimStatus>
-   <https://apisnoop.cncf.io/1.28.0/stable/core/replaceCoreV1NamespacedPersistentVolumeClaimStatus>
-   <https://apisnoop.cncf.io/1.28.0/stable/core/patchCoreV1PersistentVolumeStatus>
-   <https://apisnoop.cncf.io/1.28.0/stable/core/readCoreV1PersistentVolumeStatus>
-   <https://apisnoop.cncf.io/1.28.0/stable/core/replaceCoreV1PersistentVolumeStatus>

# API Reference and feature documentation

-   [Kubernetes API Reference Docs](https://kubernetes.io/docs/reference/kubernetes-api/)
-   [Kubernetes API / Config and Storage Resources / PersistentVolume](https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/persistent-volume-v1/)
-   [Kubernetes API / Config and Storage Resources / PersistentVolumeClaim](https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/persistent-volume-claim-v1/)
-   [client-go - PersistentVolume](https://github.com/kubernetes/client-go/tree/master/kubernetes/typed/core/v1/persistentvolume.go)
-   [client-go - PersistentVolumeClaim](https://github.com/kubernetes/client-go/tree/master/kubernetes/typed/core/v1/persistentvolumeclaim.go)

# Test outline

```
Scenario: Test the status endpoints for a pv and a pvc

  Given the e2e test has created a PVC
  When the test reads the PVC status
  Then the requested action is accepted without any error
  And the phase of the retrieved PVC status equals "Pending"

  Given the e2e test has created a PV
  When the test reads the PV status
  Then the requested action is accepted without any error
  And the phase of the retrieved PVC status equals "Available"

  Given the e2e test has created a PVC
  When the test patches the PVC status with a new condition
  Then the requested action is accepted without any error
  And the test finds the new PVC status condition with the required "patched" values

  Given the e2e test has created a PV
  When the test patches the PV status with a new "message" and "reason"
  Then the requested action is accepted without any error
  And the test finds the new "patched" values for the PV status "message" and "reason"

  Given the e2e test has patched a PVC status
  When the test updates the PVC status with a new condition
  Then the requested action is accepted without any error
  And the test finds the new PVC status condition with the required "updated" values

  Given the e2e test has patched a PV status
  When the test updates the PV status with a new "message" and "reason"
  Then the requested action is accepted without any error
  And the test finds the new "updated" values for the PV status "message" and "reason"
```

# E2E Test

Using a number of existing e2e test practices a new [ginkgo test](https://github.com/ii/kubernetes/blob/create-pv-pvc-status-test/test/e2e/storage/persistent_volumes.go#L655-L784) has been created to provide future Conformance coverage for the 6 endpoints. The e2e logs for this test are listed below.

```
[sig-storage] PersistentVolumes CSI Conformance should apply changes to a pv/pvc status
/home/ii/go/src/k8s.io/kubernetes/test/e2e/storage/persistent_volumes.go:655
  STEP: Creating a kubernetes client @ 09/26/23 11:47:17.336
  Sep 26 11:47:17.336: INFO: >>> kubeConfig: /home/ii/.kube/config
  STEP: Building a namespace api object, basename pv @ 09/26/23 11:47:17.336
  STEP: Waiting for a default service account to be provisioned in namespace @ 09/26/23 11:47:17.378
  STEP: Waiting for kube-root-ca.crt to be provisioned in namespace @ 09/26/23 11:47:17.383
  STEP: Creating initial PV and PVC @ 09/26/23 11:47:17.388
  Sep 26 11:47:17.388: INFO: Creating a PV followed by a PVC
  STEP: Listing all PVs with the labelSelector: "e2e-pv-pool=pv-6988" @ 09/26/23 11:47:17.419
  STEP: Listing PVCs in namespace "pv-6988" @ 09/26/23 11:47:17.428
  STEP: Reading "pvc-4rl4x" Status @ 09/26/23 11:47:17.435
  STEP: Reading "pv-6988-2nthc" Status @ 09/26/23 11:47:17.439
  STEP: Patching "pvc-4rl4x" Status @ 09/26/23 11:47:17.454
  STEP: Patching "pv-6988-2nthc" Status @ 09/26/23 11:47:17.465
  STEP: Updating "pvc-4rl4x" Status @ 09/26/23 11:47:17.487
  STEP: Updating "pv-6988-2nthc" Status @ 09/26/23 11:47:17.501
  Sep 26 11:47:17.523: INFO: AfterEach: deleting 1 PVCs and 1 PVs...
  Sep 26 11:47:17.524: INFO: Deleting PersistentVolumeClaim "pvc-4rl4x"
  Sep 26 11:47:17.534: INFO: Deleting PersistentVolume "pv-6988-2nthc"
  Sep 26 11:47:17.544: INFO: Waiting up to 7m0s for all (but 0) nodes to be ready
  STEP: Destroying namespace "pv-6988" for this suite. @ 09/26/23 11:47:17.55
```

# Verifying increase in coverage with APISnoop

## Listing endpoints hit by the new e2e test

This query shows the following PV and PVC endpoints are hit within a short period of running this e2e test.

```sql-mode
select distinct substring(endpoint from '\w+') AS endpoint,
                right(useragent,39) AS useragent
from testing.audit_event
where useragent like 'e2e%should%'
  and release_date::BIGINT > round(((EXTRACT(EPOCH FROM NOW()))::numeric)*1000,0) - 20000
  and endpoint ilike '%Status%'
order by endpoint
limit 10;
```

```example
                      endpoint                      |                useragent
----------------------------------------------------+-----------------------------------------
 patchCoreV1NamespacedPersistentVolumeClaimStatus   | should apply changes to a pv/pvc status
 patchCoreV1PersistentVolumeStatus                  | should apply changes to a pv/pvc status
 readCoreV1NamespacedPersistentVolumeClaimStatus    | should apply changes to a pv/pvc status
 readCoreV1PersistentVolumeStatus                   | should apply changes to a pv/pvc status
 replaceCoreV1NamespacedPersistentVolumeClaimStatus | should apply changes to a pv/pvc status
 replaceCoreV1PersistentVolumeStatus                | should apply changes to a pv/pvc status
(6 rows)

```

# Final notes

If a test with these calls gets merged, **test coverage will go up by 6 points**

This test is also created with the goal of conformance promotion.

---

/sig testing
/sig architecture
/area conformance

### PR DESCRIPTION
**Pull request by riaankleinhans**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it:**
This PR adds a test to test the following untested endpoints:
- patchCoreV1NamespacedPersistentVolumeClaimStatus
- patchCoreV1PersistentVolumeStatus
- readCoreV1NamespacedPersistentVolumeClaimStatus
- readCoreV1PersistentVolumeStatus
- replaceCoreV1NamespacedPersistentVolumeClaimStatus
- replaceCoreV1PersistentVolumeStatus

**Which issue(s) this PR fixes:**
Fixes #120891

**Testgrid Link:**
[testgrid-link](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&width=5&graph-metrics=test-duration-minutes&include-filter-by-regex=should%20apply%20changes%20to%20a%20pv/pvc%20status)

**Special notes for your reviewer:**
Adds +6 endpoint test coverage (good for conformance)

**Does this PR introduce a user-facing change?:**
```
NONE

```

**Release note:**
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:**
```
NONE

```

/sig testing
/sig architecture
/area conformance

### CODE REVIEW

**Comment by Riaankl** [comment]
/assign @carlory 
Can you please review.

**Comment by Riaankl** [comment]
/test pull-kubernetes-node-e2e-containerd

Unreleated flake
```
Summarizing 1 Failure:
  [FAIL] [sig-node] OOMKiller [LinuxOnly] [NodeConformance] init container [It] The containers terminated by OOM killer should have the reason set to OOMKilled
  test/e2e_node/oomkiller_linux_test.go:148
```

**Comment by Riaankl** [comment]
/test pull-kubernetes-e2e-gce
Unrelated issue:
```
 File "/usr/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '('kubetest', '--dump=/logs/artifacts', '--gcp-service-account=/etc/service-account/service-account.json', '--build=quick', '--stage=gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gce', '--up', '--down', '--test', '--provider=gce', '--cluster=e2e-5491c2ace1-674b9', '--gcp-network=e2e-5491c2ace1-674b9', '--extract=local', '--gcp-master-image=ubuntu', '--gcp-node-image=ubuntu', '--gcp-zone=us-west1-b', '--ginkgo-parallel=30', '--test_args=--ginkgo.skip=\\[Driver:.gcepd\\]|\\[Slow\\]|\\[Serial\\]|\\[Disruptive\\]|\\[Flaky\\]|\\[Feature:.+\\] --minStartupPods=8', '--timeout=80m')' returned non-zero exit status 1.
```

**Comment by heyste** [comment]
/test pull-kubernetes-node-e2e-containerd

<details>
<summary>unrelated flake: [sig-node]  OOMKiller [LinuxOnly] [NodeConformance] single process container The  containers terminated by OOM killer should have the reason set to  OOMKilled</summary>

```
{ failed [FAILED] pod: "oomkill-target-pod", container: "oomkill-single-target-container" has unexpected reason: "Error"
Expected
     <string>: Error
to equal
     <string>: OOMKilled
In [It] at: test/e2e_node/oomkiller_linux_test.go:148 @ 09/26/23 22:47:47.422
}
 ```
```

File: test/e2e/storage/persistent_volumes.go [hunk 1]
```diff
@@ -646,6 +651,137 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+			pvcList, err := pvcClient.List(ctx, metav1.ListOptions{})
+			framework.ExpectNoError(err, "Failed to list PVCs with the labelSelector: %q", volLabel.AsSelector().String())
+			gomega.Expect(pvcList.Items).To(gomega.HaveLen(1))
+			initialPVC := pvcList.Items[0]
+
+			ginkgo.By(fmt.Sprintf("Reading %q Status", initialPVC.Name))
+			pvcResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "persistentvolumeclaims"}
+			pvcUnstructured, err := f.DynamicClient.Resource(pvcResource).Namespace(ns).Get(ctx, initialPVC.Name, metav1.GetOptions{}, "status")
+			framework.ExpectNoError(err, "Failed to fetch the status of replica set %s in namespace %s", initialPVC.Name, ns)
+			retrievedPVC, err := unstructuredToPersistentVolumeClaim(pvcUnstructured)
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes.go:711]:
```suggestion
			retrievedPVC := &v1.PersistentVolumeClaim{}
			err = runtime.DefaultUnstructuredConverter.FromUnstructured(pvcUnstructured.Object, retrievedPVC)
```

File: test/e2e/storage/persistent_volumes.go [hunk 2]
```diff
@@ -646,6 +651,137 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+			framework.ExpectNoError(err, "Failed to retrieve %q status. %v", initialPV.Name)
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes.go:712]:
```suggestion
			framework.ExpectNoError(err, "Failed to retrieve %q status.", initialPV.Name)
```

File: test/e2e/storage/persistent_volumes.go [hunk 3]
```diff
@@ -646,6 +651,137 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+			gomega.Expect(string(retrievedPVC.Status.Phase)).To(gomega.Equal("Pending"), "Checking that the PVC status has been read")
+
+			ginkgo.By(fmt.Sprintf("Reading %q Status", initialPV.Name))
+			pvResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "persistentvolumes"}
+			pvUnstructured, err := f.DynamicClient.Resource(pvResource).Get(ctx, initialPV.Name, metav1.GetOptions{}, "status")
+			framework.ExpectNoError(err, "Failed to fetch the status of replica set %s in namespace %s", initialPV.Name, ns)
+			retrievedPV, err := unstructuredToPersistentVolume(pvUnstructured)
+			framework.ExpectNoError(err, "Failed to retrieve %q status. %v", initialPV.Name)
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes.go:720]:
```suggestion
			framework.ExpectNoError(err, "Failed to retrieve %q status.", initialPV.Name)
```

File: test/e2e/storage/persistent_volumes.go [hunk 4]
```diff
@@ -646,6 +651,137 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+			gomega.Expect(string(retrievedPV.Status.Phase)).To(gomega.Or(gomega.Equal("Available"), gomega.Equal("Bound"), gomega.Equal("Pending")), "Checking that the PV status has been read")
+
+			ginkgo.By(fmt.Sprintf("Patching %q Status", initialPVC.Name))
+			payload := []byte(`{"status":{"conditions":[{"type":"StatusPatched","status":"True", "reason":"E2E patchedStatus", "message":"Set from e2e test"}]}}`)
+
+			patchedPVC, err := pvcClient.Patch(ctx, initialPVC.Name, types.MergePatchType, payload, metav1.PatchOptions{}, "status")
+			framework.ExpectNoError(err, "Failed to patch status. %v", err)
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes.go:727]:
```suggestion
			framework.ExpectNoError(err, "Failed to patch status.")
```

File: test/e2e/storage/persistent_volumes.go [hunk 5]
```diff
@@ -646,6 +651,137 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+					"Message": gomega.ContainSubstring("Set from e2e test"),
+					"Reason":  gomega.ContainSubstring("E2E patchedStatus"),
+				}),
+			}), "Checking that patched status has been applied")
+
+			ginkgo.By(fmt.Sprintf("Patching %q Status", retrievedPV.Name))
+			payload = []byte(`{"status":{"message": "StatusPatched", "reason": "E2E patchStatus"}}`)
+
+			patchedPV, err := pvClient.Patch(ctx, retrievedPV.Name, types.MergePatchType, payload, metav1.PatchOptions{}, "status")
+			framework.ExpectNoError(err, "Failed to patch %q status. %v", retrievedPV.Name, err)
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes.go:740]:
```suggestion
			framework.ExpectNoError(err, "Failed to patch %q status.", retrievedPV.Name)
```

File: test/e2e/storage/persistent_volumes.go [hunk 6]
```diff
@@ -646,6 +651,137 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+			framework.ExpectNoError(err, "Failed to fetch the status of replica set %s in namespace %s", initialPVC.Name, ns)
+			retrievedPVC, err := unstructuredToPersistentVolumeClaim(pvcUnstructured)
+			framework.ExpectNoError(err, "Failed to retrieve %q status. %v", initialPV.Name)
+			gomega.Expect(string(retrievedPVC.Status.Phase)).To(gomega.Equal("Pending"), "Checking that the PVC status has been read")
+
+			ginkgo.By(fmt.Sprintf("Reading %q Status", initialPV.Name))
+			pvResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "persistentvolumes"}
+			pvUnstructured, err := f.DynamicClient.Resource(pvResource).Get(ctx, initialPV.Name, metav1.GetOptions{}, "status")
+			framework.ExpectNoError(err, "Failed to fetch the status of replica set %s in namespace %s", initialPV.Name, ns)
+			retrievedPV, err := unstructuredToPersistentVolume(pvUnstructured)
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes.go:719]:
```suggestion
			retrievedPV := &v1.PersistentVolume{}
			err = runtime.DefaultUnstructuredConverter.FromUnstructured(pvUnstructured.Object, retrievedPV)
```

File: test/e2e/storage/persistent_volumes.go [hunk 7]
```diff
@@ -646,6 +651,137 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+					Type:    "StatusUpdated",
+					Status:  "True",
+					Reason:  "E2E updateStatus",
+					Message: "Set from e2e test",
+				})
+
+				updatedPVC, err = pvcClient.UpdateStatus(ctx, statusToUpdate, metav1.UpdateOptions{})
+				return err
+			})
+			framework.ExpectNoError(err, "Failed to update status. %v", err)
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes.go:761]:
```suggestion
			framework.ExpectNoError(err, "Failed to update status.")
```

File: test/e2e/storage/persistent_volumes.go [hunk 8]
```diff
@@ -646,6 +651,137 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				pvToUpdate, err = pvClient.Get(ctx, patchedPV.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err, "Unable to retrieve pv %s", patchedPV.Name)
+
+				pvToUpdate.Status.Reason = "E2E updateStatus"
+				pvToUpdate.Status.Message = "StatusUpdated"
+				updatedPV, err = pvClient.UpdateStatus(ctx, pvToUpdate, metav1.UpdateOptions{})
+				return err
+			})
+			framework.ExpectNoError(err, "Failed to update status. %v", err)
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes.go:781]:
```suggestion
			framework.ExpectNoError(err, "Failed to update status.")
```

**Comment by heyste** [comment]
/test pull-kubernetes-e2e-gce

<details>
<summary> Unrelated flake</summary>

```
Init container clonerefs not ready: (state: terminated, reason: "ContainerStatusUnknown", message: "The container could not be located when the pod was terminated") Init container initupload not ready: (state: waiting, reason: "PodInitializing", message: "") Init container place-entrypoint not ready: (state: waiting, reason: "PodInitializing", message: "")
```
</details>

File: test/e2e/storage/persistent_volumes.go (hunk 1 above)
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Fixed

File: test/e2e/storage/persistent_volumes.go (hunk 2 above)
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Fixed

File: test/e2e/storage/persistent_volumes.go (hunk 3 above)
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Fixed

File: test/e2e/storage/persistent_volumes.go (hunk 4 above)
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Fixed

File: test/e2e/storage/persistent_volumes.go (hunk 5 above)
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Fixed

File: test/e2e/storage/persistent_volumes.go (hunk 6 above)
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Fixed

File: test/e2e/storage/persistent_volumes.go (hunk 7 above)
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Fixed

File: test/e2e/storage/persistent_volumes.go (hunk 8 above)
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Fixed

File: test/e2e/storage/persistent_volumes.go [hunk 9]
```diff
@@ -646,6 +649,139 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+
+			ginkgo.By(fmt.Sprintf("Listing PVCs in namespace %q", ns))
+			pvcList, err := pvcClient.List(ctx, metav1.ListOptions{})
+			framework.ExpectNoError(err, "Failed to list PVCs with the labelSelector: %q", volLabel.AsSelector().String())
+			gomega.Expect(pvcList.Items).To(gomega.HaveLen(1))
+			initialPVC := pvcList.Items[0]
+
+			ginkgo.By(fmt.Sprintf("Reading %q Status", initialPVC.Name))
+			pvcResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "persistentvolumeclaims"}
+			pvcUnstructured, err := f.DynamicClient.Resource(pvcResource).Namespace(ns).Get(ctx, initialPVC.Name, metav1.GetOptions{}, "status")
```
Comment (msau42) [inline test/e2e/storage/persistent_volumes.go:707]: I'm curious why we need to use the dynamic client?

Comment (heyste) [reply test/e2e/storage/persistent_volumes.go:707]: AFAIK the only way to read any subresource is to use the dynamic client. Here's an example of reading the [status for a network service](https://github.com/kubernetes/kubernetes/blob/3fc8c324256593db043601cfd507c76d54b07802/test/e2e/network/service.go#L3388)

File: test/e2e/storage/persistent_volumes.go [hunk 10]
```diff
@@ -646,6 +649,139 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+			framework.ExpectNoError(err, "Failed to fetch the status of replica set %s in namespace %s", initialPVC.Name, ns)
+			retrievedPVC := &v1.PersistentVolumeClaim{}
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(pvcUnstructured.UnstructuredContent(), &retrievedPVC)
+			framework.ExpectNoError(err, "Failed to retrieve %q status.", initialPV.Name)
+			gomega.Expect(string(retrievedPVC.Status.Phase)).To(gomega.Equal("Pending"), "Checking that the PVC status has been read")
+
+			ginkgo.By(fmt.Sprintf("Reading %q Status", initialPV.Name))
+			pvResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "persistentvolumes"}
+			pvUnstructured, err := f.DynamicClient.Resource(pvResource).Get(ctx, initialPV.Name, metav1.GetOptions{}, "status")
+			framework.ExpectNoError(err, "Failed to fetch the status of replica set %s in namespace %s", initialPV.Name, ns)
```
Comment (jsafrane) [inline test/e2e/storage/persistent_volumes.go:717]:
```suggestion
			framework.ExpectNoError(err, "Failed to fetch the status of PV %s in namespace %s", initialPV.Name, ns)
```

File: test/e2e/storage/persistent_volumes.go [hunk 11]
```diff
@@ -646,6 +649,139 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+					Labels: csiDriverLabel,
+				},
+
+				Spec: storagev1.CSIDriverSpec{
+					VolumeLifecycleModes: []storagev1.VolumeLifecycleMode{
+						storagev1.VolumeLifecyclePersistent,
+					},
+					FSGroupPolicy: &defaultFSGroupPolicy,
+				},
+			}
```
[the remark this answers is not in the document]
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Thanks for the info. Fixed

File: test/e2e/storage/persistent_volumes.go [hunk 12]
```diff
@@ -646,6 +649,139 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+
+			pvNamePrefix := ns + "-"
+			pvHostPathConfig := e2epv.PersistentVolumeConfig{
+				NamePrefix: pvNamePrefix,
+				Labels:     volLabel,
```
[the remark this answers is not in the document]
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Fixed

File: test/e2e/storage/persistent_volumes.go [hunk 13]
```diff
@@ -646,6 +649,139 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
...
+			ginkgo.By(fmt.Sprintf("Listing PVCs in namespace %q", ns))
+			pvcList, err := pvcClient.List(ctx, metav1.ListOptions{})
+			framework.ExpectNoError(err, "Failed to list PVCs with the labelSelector: %q", volLabel.AsSelector().String())
+			gomega.Expect(pvcList.Items).To(gomega.HaveLen(1))
+			initialPVC := pvcList.Items[0]
+
+			ginkgo.By(fmt.Sprintf("Reading %q Status", initialPVC.Name))
+			pvcResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "persistentvolumeclaims"}
+			pvcUnstructured, err := f.DynamicClient.Resource(pvcResource).Namespace(ns).Get(ctx, initialPVC.Name, metav1.GetOptions{}, "status")
+			framework.ExpectNoError(err, "Failed to fetch the status of replica set %s in namespace %s", initialPVC.Name, ns)
```
[the remark this answers is not in the document]
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Thanks for catching this. Fixed

File: test/e2e/storage/persistent_volumes.go (hunk 10 above)
Comment (heyste) [reply test/e2e/storage/persistent_volumes.go]: Fixed

**Comment by Riaankl** [comment]
Good catch on your review @jsafrane 
We fixed all the nits.
Can you please check and `/lgtm` & `/approve`

Thanks!

**Pull request merged**